src/main/groovy/com/twcable/grabbit/jcr/JCRNodeDecorator.groovy

```diff
@@ -105,6 +106,37 @@ class JCRNodeDecorator {
         return definition.isMandatory()
     }
 
+    void checkoutIfNecessary(javax.jcr.Node node) throws RepositoryException {
```

Comment:
This isn't something we should expose. If we are writing to a node, we should capture that within the decorator, and checkout appropriately

Rough pseudo code : 

```groovy
setProperty(name, value, type) {
  modify {
    setProperty(name, value, type
  }
}

modify(Closure<Void> doModify) {
  if(isVersionable()) {
    checkout()
    doModify.call()
    checkin()
  }
  else {
    doModify()
  }
}